cli/bpmetadata/proto/bpmetadata.proto

```diff
@@ -0,0 +1,332 @@
+    BlueprintUIOutput runtime = 2;
+  }
+}
+
+message BlueprintRepoDetail {
+  // Autogenerated: URL from the .git dir.
+  // Can be manually overriden with a custom URL if needed.
+  string repo = 1;
+
+  // Set as "git" for now until more types are supported.
+  string source_type = 2;
+}
+
+// BlueprintActuationTool defines the actuation tool used to provision the blueprint.
+message BlueprintActuationTool {
+  // Set as "Terraform" for now until more flavors are supported.
+  string flavor = 1;
+
+  // Required version for the actuation tool.
+  // Autogenerated: For Terraform this is the `required_version` set in
+  // `terraform` block. E.g.
+  // terraform {
+  //   required_version = ">= 0.13"
+  // }
+  string version = 2;
+}
+
+// BlueprintDescription defines the description of the blueprint.
+message BlueprintDescription {
+  // Autogenerated: All types of descriptions are set with the markdown content
+  // immediately after the each type's heading declaration in readme.md.
+
+  // Autogenerated: Markdown after "### Tagline".
+  string tagline = 1;
+
+  // Autogenerated: Markdown after "### Detailed".
+  string detailed = 2;
+
+  // Autogenerated: Markdown after "### PreDeploy".
+  string preDeploy = 3;
+
+  // Autogenerated: Markdown after "### Html".
+  string html = 4;
+
+  // Autogenerated: Markdown after "### EulaUrls".
+  repeated string eulaURLs = 5;
+
+  // Autogenerated: Markdown after "### Architecture"
+  // Deprecated. Use BlueprintContent.Architecture instead.
+  repeated string architecture = 6;
+}
+
+// A time estimate in secs required for configuring and deploying the blueprint.
+message BlueprintTimeEstimate {
+  // Autogenerated: Set using the content defined under "### DeploymentTime" E.g.
+  // ### DeploymentTime
+  // - Configuration: X secs
+  // - Deployment: Y secs
+  int32 configurationSecs = 1;
+  int32 deploymentSecs = 2;
+}
+
+// The cost estimate for the blueprint based on preconfigured variables.
+message BlueprintCostEstimate {
+  // Autogenerated: Set using the content defined under "### Cost" as a link
+  // with a description E.g.
+  // ### Cost
+  // [View cost details](https://cloud.google.com/products/calculator?hl=en_US&_ga=2.1665458.-226505189.1675191136#id=02fb0c45-cc29-4567-8cc6-f72ac9024add)
+  string description = 1;
+  string url = 2;
+}
+
+// A GCP cloud product used in the blueprint.
+// Manually authored.
+message BlueprintCloudProduct {
+  // A top-level (e.g. "Compute Engine") or secondary (e.g. "Binary Authorization")
+  // product used in the blueprint.
+  string productId = 1;
+
+  // Url for the product.
+  string pageUrl = 2;
+
+  // A label string for the product, if it is not an integrated GCP product.
+  // E.g. "Data Studio"
+  string label = 3;
+}
+
+// QuotaResourceType defines the type of resource a quota is applied to.
+enum QuotaResourceType {
+  GCE_INSTANCE = 0;
+  GCE_DISK = 1;
+}
+
+// // QuotaType defines the type of quota.
+// enum QuotaType {
```

Comment:
commented?

Reply:
Yeah because protos don't allow enums for keys in map types. I'll take this out.